src/main/webapp/scripts/index.js

```diff
@@ -2,13 +2,19 @@ $().ready(function() {
   const location = getCookie('location');
   if (location != "") {
     $('#location-checkbox').prop('checked', true);
+    setCoordinateCookie();
+  }
+  else {
+    eraseCookie('coordinates');
```

Comment:
Where are you setting up this cookie? I think you should be erasing lat and lng instead.